src/components/app-link/app-link.vue

```diff
@@ -9,9 +9,18 @@
   });
   const router = useRouter();
 
-  const normalizedTo = computed(() =>  withTrailingSlash(
-    typeof props.to === 'object' ? router.resolve(props.to).path : props.to
+  const resolvedRoute = computed(() => router.resolve(props.to));
+  const normalizedPath = computed(() =>  withTrailingSlash(
+    typeof props.to === 'object' ? resolvedRoute.value.path : props.to
   ));
+  const normalizedTo = computed(() =>
+    typeof props.to === "object"
+      ? {
+          ...resolvedRoute.value,
+          path: normalizedPath.value,
+        }
+      : normalizedPath.value
+  );
```

Comment:
Normalizing only the path means that `hash` isn't ignored.
Using the `resolvedRoute` on `normalizedTo` leads to having a named route and prevents this error `was passed with params but they will be ignored. Use a named route alongside params instead.` (you can see the warnings on the console if you checkout this commit 9f335fbe6cf31a49c1924b34e11ba11d3f9f4812.

Another approach would be to send a string instead of an object from the `pagination-nav` then it would have to receive a `basePath` as a prop, but I didn't find a nice way of getting that base path that wasn't just hardcoding it. Let me know if you prefer this approach and have suggestions on how to implement it.